active/PagerControl/PagerControl.md

```diff
@@ -0,0 +1,380 @@
+                DisplayMode="ComboBox"
+                NumberOfPages="10"
+                NextButtonVisibility="HiddenOnEdge"
+                LastButtonVisibility="None"/>
+        </muxc:PagerControl>
+</Grid>
+```
+
+### Code Example
+
+Below is a code sample for how to use the pager control with a data grid and how to handle the page changed event. Since this control is only a UI component, the data managment still needs to be handled through code. This example also shows how to set the number of pages property and manage what images need to be shown to the user when they select a specific page number. 
+
+When a user interacts with the pager control and selects an edge button or page number, the page changed event will fire. This event will provide the index the user is coming from (PreviousPageIndex) and the index the user is navigating to (NewPageIndex). 
+
+XAML
+```XAML
+<Page
+    x:Class="WindowsGallaryApp.Samples.DataGridSamplePage"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:WindowsGallaryApp.Samples"
+    xmlns:controls="using:Microsoft.Toolkit.Uwp.UI.Controls"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    xmlns:muxc="using:Microsoft.UI.Xaml.Controls"
+    mc:Ignorable="d"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+    <Page.Resources>
+        <CollectionViewSource x:Name="imagesCVS" />
+        <DataTemplate x:Key="ImageTemplate" x:DataType="CustomImage">
+            <Image Stretch="UniformToFill" Source="{x:Bind ImageLocation}"
+                   Width="190" Height="130" />
+        </DataTemplate>
+    </Page.Resources>
+
+    <Grid>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="10*"/>
+            <RowDefinition Height="Auto"/>
+        </Grid.RowDefinitions>
+        <TextBlock Grid.Row="0"
+                   Text="Image Gallery - DataGrid"
+                   Margin="10,0,0,0"
+                   Style="{ThemeResource TitleTextBlockStyle}"/>
+        <controls:DataGrid Grid.Row="1" 
+                           ItemsSource="{x:Bind imagesCVS.View, Mode=OneWay}"
+                           AutoGenerateColumns="True"
+                           AutoGeneratingColumn="DataGrid_AutoGeneratingColumn"/>
+        <data:Pager x:Name="Pager" Grid.Row="2" PageChanged="Pager_PageChanged" PagerDisplayMode="ButtonPanel"/>
+    </Grid>
+</Page>
+```
+
+C#
+```C#
+        private List<CustomImage> imageList;
+        private int imagesPerPage = 25;
+
+        public DataGridSamplePage()
+        {
+                this.InitializeComponent();
+                Loaded += OnPageLoaded;
+        }
+
+        private async void OnPageLoaded(object sender, RoutedEventArgs e)
+        {
+                imageList = await GetImageList();
```

Comment:
Per an earlier comment, I think the primary scenario for this control is data virtualization. Towards that end, to make this sample more realistic, it shouldn't show the images all being pre-fetched; it should show a call to fetch images when the page changes. And then to continue the realism, it needs to show an indeterminate progress bar during the async call to fetch the images from a server?

Reply:
@MikeHillberg in the Pager_PageChanged method we do call GetPageImages to update the images being shown when the page number changes. OnPageLoaded is only called on the initial load. Am I understanding your comment correctly? Are you saying that the imageList = await GetImageList(); should not be included and the images should be retrieved adhoc?

Reply:
Agree. We could just set the source to a set of images fetched from somewhere based on the page number. That could be a web call for example.

Reply:
@MarissaMatt - Right, in PageChanged its extracting some images. But in PageLoaded it's getting all of the images ready, probably downloading them all from a server. Pre-downloading all the images like that defeat the purpose of the pager, you might as well put them in a ScrollViewer.

Reply:
I updated the sample in the previous comment to do an async call with just the images fetched for each page.